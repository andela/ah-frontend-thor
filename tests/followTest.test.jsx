/* eslint-disable no-shadow */
/* eslint-disable react/jsx-no-undef */
import { mount, shallow } from "enzyme";

import React from "react";
import expect from "expect";
import { MemoryRouter } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import Follow, {RelatedArticles, FollowButton, UnFollowButton, mapStateToProps, mapDispatchToProps} from "../src/components/Follow";
import FollowStatus from "../src/components/FollowStatus";


const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({ followStatusReducer: {results: []},
  followUser: jest.fn(),
  getFollowStatus: jest.fn(),
  unfollowUser: jest.fn(),
});

it("it should render component", () => {
  shallow(
    <MemoryRouter>
      <Provider store={store}><Follow /></Provider>
    </MemoryRouter>,
  );
});

const props = {
  auth: {
    "errors": "User with this name or password not found !"
  },
  onClick: jest.fn(),
  loginUser: jest.fn(),
  followedUsername: "cdvx",
};

describe("<Follow />", () => {
  test("renders the component", () => {
    const wrapper = shallow(<Provider store={store}><Follow /></Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<RelatedArticles />", () => {
  test("renders the component", () => {
    const wrapper = shallow(<RelatedArticles />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<FollowButton />", () => {
  test("renders the component", () => {
    const wrapper = shallow(<FollowButton onClick={props.onClick} followedUsername={props.followedUsername} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<UnFollowButton />", () => {
  test("renders the component", () => {
    const wrapper = shallow(<UnFollowButton onClick={props.onClick} followedUsername={props.followedUsername} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<FollowStatus />", () => {
  test("renders the component", () => {
    const wrapper = shallow(<Provider store={store}><FollowStatus /></Provider>);

    expect(wrapper).toMatchSnapshot();
  });
});


describe("test the login container", () => {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const storee = mockStore({ followStatusReducer: {results: []}, followUser: jest.fn(), getFollowStatus: jest.fn(), unfollowUser:jest.fn() });
  const fakeEvent = { preventDefault: () => {}};

  const wrapper = mount(<Provider store={storee}><Follow fakeEvent /></Provider>);
  wrapper.setState({
    followedByUser: [],
    followingUsername: "chucky",
    followedUsername: "cdvx"
  });

  wrapper.setProps({followUser: jest.fn(), getFollowStatus: jest.fn(), unfollowUser:jest.fn() });
  it("should mount the login component", () => {

    expect(wrapper.find("FollowButton").simulate("click", fakeEvent));
    // wrapper.find("FollowButton").simulate("click", fakeEvent);

  });
});

describe("test the Follow container", () => {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const storee = mockStore({ followStatusReducer: {results: [{}]}, followUser: jest.fn(), getFollowStatus: jest.fn(), unfollowUser:jest.fn() });
  const fakeEvent = { preventDefault: () => {}};

  const wrapper = mount(<Provider store={storee}><Follow fakeEvent /></Provider>);
  wrapper.setState({
    followedByUser: [],
    followingUsername: "chucky",
    followedUsername: "cdvx"
  });

  wrapper.setProps({followUser: jest.fn(), getFollowStatus: jest.fn(), unfollowUser:jest.fn() });
  it("should mount the login component", () => {

    expect(wrapper.find("UnFollowButton").simulate("click", fakeEvent));

  });
});

describe("Login", () => {
  it("should show previously previously passed state", () => {
    const state = {
      followStatusReducer: {results: []}
    };
    expect(mapStateToProps(state)).toEqual({results: []});

  });

  it("Should pass action to props on submit", () => {
    const dispatch = jest.fn();
    const fakeEvent = { preventDefault: () => {}};
    mapDispatchToProps(dispatch).followUser(fakeEvent, "cdvx");
    expect((dispatch.mock.calls[0][0])).toBeTruthy();
    mapDispatchToProps(dispatch).getFollowStatus(fakeEvent, "cdvx");
    expect((dispatch.mock.calls[0][0])).toBeTruthy();
    mapDispatchToProps(dispatch).unfollowUser(fakeEvent, "cdvx");
    expect((dispatch.mock.calls[0][0])).toBeTruthy();
  });
});


describe("<UnFollowButton />", () => {
  describe("onClick()", () => {
    test("successfully calls the onSubmit handler", () => {
      const wrapper = shallow(
        <UnFollowButton onClick={props.onClick} />
      );
      wrapper.simulate("onClick");

      expect(props.onClick).toBeTruthy();
    });
  });
});


describe("Checks whether mapstate to props returns", () => {
  const expectedProp = {
    followStatusReducer: {results: []},
    followUser: jest.fn(),
    getFollowStatus: jest.fn(),
    unfollowUser: jest.fn(),
  };

  expect(mapStateToProps(expectedProp)).toBeTruthy();
});


describe("Checks whether mapdispatch to props returns", () => {
  const dispatch = jest.fn();
  mapDispatchToProps(dispatch).followUser();
  expect(dispatch.mock.calls.length).toBe(1);
});
