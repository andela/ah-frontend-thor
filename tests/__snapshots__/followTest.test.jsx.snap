// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Follow /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(Follow) />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<FollowButton /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FollowButton
    followedUsername="cdvx"
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="btn  btn-outline-blue btn-lg"
        href="#"
      >
        Follow 
        cdvx
      </a>,
      "className": "blog-pagination",
      "id": "follow",
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Follow ",
          "cdvx",
        ],
        "className": "btn  btn-outline-blue btn-lg",
        "href": "#",
      },
      "ref": null,
      "rendered": Array [
        "Follow ",
        "cdvx",
      ],
      "type": "a",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="btn  btn-outline-blue btn-lg"
          href="#"
        >
          Follow 
          cdvx
        </a>,
        "className": "blog-pagination",
        "id": "follow",
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Follow ",
            "cdvx",
          ],
          "className": "btn  btn-outline-blue btn-lg",
          "href": "#",
        },
        "ref": null,
        "rendered": Array [
          "Follow ",
          "cdvx",
        ],
        "type": "a",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<FollowStatus /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(FollowStatus) />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<RelatedArticles /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RelatedArticles />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="list-unstyled"
      >
        <li
          className="media"
        >
          <img
            alt="Generic placeholder image"
            className="d-flex mr-3"
            src="https://mdbootstrap.com/img/Photos/Others/placeholder7.jpg"
          />
          <div
            className="media-body"
          >
            <a
              href=""
            >
              <h5
                className="mt-0 mb-1 font-weight-bold"
              >
                List-based media object
              </h5>
            </a>
            Cras sit amet nibh libero, in gravida nulla (...)
          </div>
        </li>
        <li
          className="media my-4"
        >
          <img
            alt="An image"
            className="d-flex mr-3"
            src="https://mdbootstrap.com/img/Photos/Others/placeholder6.jpg"
          />
          <div
            className="media-body"
          >
            <a
              href=""
            >
              <h5
                className="mt-0 mb-1 font-weight-bold"
              >
                List-based media object
              </h5>
            </a>
            Cras sit amet nibh libero, in gravida nulla (...)
          </div>
        </li>
        <li
          className="media"
        >
          <img
            alt="Generic placeholder image"
            className="d-flex mr-3"
            src="https://mdbootstrap.com/img/Photos/Others/placeholder5.jpg"
          />
          <div
            className="media-body"
          >
            <a
              href=""
            >
              <h5
                className="mt-0 mb-1 font-weight-bold"
              >
                List-based media object
              </h5>
            </a>
            Cras sit amet nibh libero, in gravida nulla (...)
          </div>
        </li>
      </ul>,
      "className": "card-body",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="media"
          >
            <img
              alt="Generic placeholder image"
              className="d-flex mr-3"
              src="https://mdbootstrap.com/img/Photos/Others/placeholder7.jpg"
            />
            <div
              className="media-body"
            >
              <a
                href=""
              >
                <h5
                  className="mt-0 mb-1 font-weight-bold"
                >
                  List-based media object
                </h5>
              </a>
              Cras sit amet nibh libero, in gravida nulla (...)
            </div>
          </li>,
          <li
            className="media my-4"
          >
            <img
              alt="An image"
              className="d-flex mr-3"
              src="https://mdbootstrap.com/img/Photos/Others/placeholder6.jpg"
            />
            <div
              className="media-body"
            >
              <a
                href=""
              >
                <h5
                  className="mt-0 mb-1 font-weight-bold"
                >
                  List-based media object
                </h5>
              </a>
              Cras sit amet nibh libero, in gravida nulla (...)
            </div>
          </li>,
          <li
            className="media"
          >
            <img
              alt="Generic placeholder image"
              className="d-flex mr-3"
              src="https://mdbootstrap.com/img/Photos/Others/placeholder5.jpg"
            />
            <div
              className="media-body"
            >
              <a
                href=""
              >
                <h5
                  className="mt-0 mb-1 font-weight-bold"
                >
                  List-based media object
                </h5>
              </a>
              Cras sit amet nibh libero, in gravida nulla (...)
            </div>
          </li>,
        ],
        "className": "list-unstyled",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Generic placeholder image"
                className="d-flex mr-3"
                src="https://mdbootstrap.com/img/Photos/Others/placeholder7.jpg"
              />,
              <div
                className="media-body"
              >
                <a
                  href=""
                >
                  <h5
                    className="mt-0 mb-1 font-weight-bold"
                  >
                    List-based media object
                  </h5>
                </a>
                Cras sit amet nibh libero, in gravida nulla (...)
              </div>,
            ],
            "className": "media",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Generic placeholder image",
                "className": "d-flex mr-3",
                "src": "https://mdbootstrap.com/img/Photos/Others/placeholder7.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    href=""
                  >
                    <h5
                      className="mt-0 mb-1 font-weight-bold"
                    >
                      List-based media object
                    </h5>
                  </a>,
                  "Cras sit amet nibh libero, in gravida nulla (...)",
                ],
                "className": "media-body",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h5
                      className="mt-0 mb-1 font-weight-bold"
                    >
                      List-based media object
                    </h5>,
                    "href": "",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "List-based media object",
                      "className": "mt-0 mb-1 font-weight-bold",
                    },
                    "ref": null,
                    "rendered": "List-based media object",
                    "type": "h5",
                  },
                  "type": "a",
                },
                "Cras sit amet nibh libero, in gravida nulla (...)",
              ],
              "type": "div",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="An image"
                className="d-flex mr-3"
                src="https://mdbootstrap.com/img/Photos/Others/placeholder6.jpg"
              />,
              <div
                className="media-body"
              >
                <a
                  href=""
                >
                  <h5
                    className="mt-0 mb-1 font-weight-bold"
                  >
                    List-based media object
                  </h5>
                </a>
                Cras sit amet nibh libero, in gravida nulla (...)
              </div>,
            ],
            "className": "media my-4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "An image",
                "className": "d-flex mr-3",
                "src": "https://mdbootstrap.com/img/Photos/Others/placeholder6.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    href=""
                  >
                    <h5
                      className="mt-0 mb-1 font-weight-bold"
                    >
                      List-based media object
                    </h5>
                  </a>,
                  "Cras sit amet nibh libero, in gravida nulla (...)",
                ],
                "className": "media-body",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h5
                      className="mt-0 mb-1 font-weight-bold"
                    >
                      List-based media object
                    </h5>,
                    "href": "",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "List-based media object",
                      "className": "mt-0 mb-1 font-weight-bold",
                    },
                    "ref": null,
                    "rendered": "List-based media object",
                    "type": "h5",
                  },
                  "type": "a",
                },
                "Cras sit amet nibh libero, in gravida nulla (...)",
              ],
              "type": "div",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Generic placeholder image"
                className="d-flex mr-3"
                src="https://mdbootstrap.com/img/Photos/Others/placeholder5.jpg"
              />,
              <div
                className="media-body"
              >
                <a
                  href=""
                >
                  <h5
                    className="mt-0 mb-1 font-weight-bold"
                  >
                    List-based media object
                  </h5>
                </a>
                Cras sit amet nibh libero, in gravida nulla (...)
              </div>,
            ],
            "className": "media",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Generic placeholder image",
                "className": "d-flex mr-3",
                "src": "https://mdbootstrap.com/img/Photos/Others/placeholder5.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    href=""
                  >
                    <h5
                      className="mt-0 mb-1 font-weight-bold"
                    >
                      List-based media object
                    </h5>
                  </a>,
                  "Cras sit amet nibh libero, in gravida nulla (...)",
                ],
                "className": "media-body",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h5
                      className="mt-0 mb-1 font-weight-bold"
                    >
                      List-based media object
                    </h5>,
                    "href": "",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "List-based media object",
                      "className": "mt-0 mb-1 font-weight-bold",
                    },
                    "ref": null,
                    "rendered": "List-based media object",
                    "type": "h5",
                  },
                  "type": "a",
                },
                "Cras sit amet nibh libero, in gravida nulla (...)",
              ],
              "type": "div",
            },
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="list-unstyled"
        >
          <li
            className="media"
          >
            <img
              alt="Generic placeholder image"
              className="d-flex mr-3"
              src="https://mdbootstrap.com/img/Photos/Others/placeholder7.jpg"
            />
            <div
              className="media-body"
            >
              <a
                href=""
              >
                <h5
                  className="mt-0 mb-1 font-weight-bold"
                >
                  List-based media object
                </h5>
              </a>
              Cras sit amet nibh libero, in gravida nulla (...)
            </div>
          </li>
          <li
            className="media my-4"
          >
            <img
              alt="An image"
              className="d-flex mr-3"
              src="https://mdbootstrap.com/img/Photos/Others/placeholder6.jpg"
            />
            <div
              className="media-body"
            >
              <a
                href=""
              >
                <h5
                  className="mt-0 mb-1 font-weight-bold"
                >
                  List-based media object
                </h5>
              </a>
              Cras sit amet nibh libero, in gravida nulla (...)
            </div>
          </li>
          <li
            className="media"
          >
            <img
              alt="Generic placeholder image"
              className="d-flex mr-3"
              src="https://mdbootstrap.com/img/Photos/Others/placeholder5.jpg"
            />
            <div
              className="media-body"
            >
              <a
                href=""
              >
                <h5
                  className="mt-0 mb-1 font-weight-bold"
                >
                  List-based media object
                </h5>
              </a>
              Cras sit amet nibh libero, in gravida nulla (...)
            </div>
          </li>
        </ul>,
        "className": "card-body",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="media"
            >
              <img
                alt="Generic placeholder image"
                className="d-flex mr-3"
                src="https://mdbootstrap.com/img/Photos/Others/placeholder7.jpg"
              />
              <div
                className="media-body"
              >
                <a
                  href=""
                >
                  <h5
                    className="mt-0 mb-1 font-weight-bold"
                  >
                    List-based media object
                  </h5>
                </a>
                Cras sit amet nibh libero, in gravida nulla (...)
              </div>
            </li>,
            <li
              className="media my-4"
            >
              <img
                alt="An image"
                className="d-flex mr-3"
                src="https://mdbootstrap.com/img/Photos/Others/placeholder6.jpg"
              />
              <div
                className="media-body"
              >
                <a
                  href=""
                >
                  <h5
                    className="mt-0 mb-1 font-weight-bold"
                  >
                    List-based media object
                  </h5>
                </a>
                Cras sit amet nibh libero, in gravida nulla (...)
              </div>
            </li>,
            <li
              className="media"
            >
              <img
                alt="Generic placeholder image"
                className="d-flex mr-3"
                src="https://mdbootstrap.com/img/Photos/Others/placeholder5.jpg"
              />
              <div
                className="media-body"
              >
                <a
                  href=""
                >
                  <h5
                    className="mt-0 mb-1 font-weight-bold"
                  >
                    List-based media object
                  </h5>
                </a>
                Cras sit amet nibh libero, in gravida nulla (...)
              </div>
            </li>,
          ],
          "className": "list-unstyled",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="Generic placeholder image"
                  className="d-flex mr-3"
                  src="https://mdbootstrap.com/img/Photos/Others/placeholder7.jpg"
                />,
                <div
                  className="media-body"
                >
                  <a
                    href=""
                  >
                    <h5
                      className="mt-0 mb-1 font-weight-bold"
                    >
                      List-based media object
                    </h5>
                  </a>
                  Cras sit amet nibh libero, in gravida nulla (...)
                </div>,
              ],
              "className": "media",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Generic placeholder image",
                  "className": "d-flex mr-3",
                  "src": "https://mdbootstrap.com/img/Photos/Others/placeholder7.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <a
                      href=""
                    >
                      <h5
                        className="mt-0 mb-1 font-weight-bold"
                      >
                        List-based media object
                      </h5>
                    </a>,
                    "Cras sit amet nibh libero, in gravida nulla (...)",
                  ],
                  "className": "media-body",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h5
                        className="mt-0 mb-1 font-weight-bold"
                      >
                        List-based media object
                      </h5>,
                      "href": "",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "List-based media object",
                        "className": "mt-0 mb-1 font-weight-bold",
                      },
                      "ref": null,
                      "rendered": "List-based media object",
                      "type": "h5",
                    },
                    "type": "a",
                  },
                  "Cras sit amet nibh libero, in gravida nulla (...)",
                ],
                "type": "div",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="An image"
                  className="d-flex mr-3"
                  src="https://mdbootstrap.com/img/Photos/Others/placeholder6.jpg"
                />,
                <div
                  className="media-body"
                >
                  <a
                    href=""
                  >
                    <h5
                      className="mt-0 mb-1 font-weight-bold"
                    >
                      List-based media object
                    </h5>
                  </a>
                  Cras sit amet nibh libero, in gravida nulla (...)
                </div>,
              ],
              "className": "media my-4",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "An image",
                  "className": "d-flex mr-3",
                  "src": "https://mdbootstrap.com/img/Photos/Others/placeholder6.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <a
                      href=""
                    >
                      <h5
                        className="mt-0 mb-1 font-weight-bold"
                      >
                        List-based media object
                      </h5>
                    </a>,
                    "Cras sit amet nibh libero, in gravida nulla (...)",
                  ],
                  "className": "media-body",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h5
                        className="mt-0 mb-1 font-weight-bold"
                      >
                        List-based media object
                      </h5>,
                      "href": "",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "List-based media object",
                        "className": "mt-0 mb-1 font-weight-bold",
                      },
                      "ref": null,
                      "rendered": "List-based media object",
                      "type": "h5",
                    },
                    "type": "a",
                  },
                  "Cras sit amet nibh libero, in gravida nulla (...)",
                ],
                "type": "div",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="Generic placeholder image"
                  className="d-flex mr-3"
                  src="https://mdbootstrap.com/img/Photos/Others/placeholder5.jpg"
                />,
                <div
                  className="media-body"
                >
                  <a
                    href=""
                  >
                    <h5
                      className="mt-0 mb-1 font-weight-bold"
                    >
                      List-based media object
                    </h5>
                  </a>
                  Cras sit amet nibh libero, in gravida nulla (...)
                </div>,
              ],
              "className": "media",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Generic placeholder image",
                  "className": "d-flex mr-3",
                  "src": "https://mdbootstrap.com/img/Photos/Others/placeholder5.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <a
                      href=""
                    >
                      <h5
                        className="mt-0 mb-1 font-weight-bold"
                      >
                        List-based media object
                      </h5>
                    </a>,
                    "Cras sit amet nibh libero, in gravida nulla (...)",
                  ],
                  "className": "media-body",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h5
                        className="mt-0 mb-1 font-weight-bold"
                      >
                        List-based media object
                      </h5>,
                      "href": "",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "List-based media object",
                        "className": "mt-0 mb-1 font-weight-bold",
                      },
                      "ref": null,
                      "rendered": "List-based media object",
                      "type": "h5",
                    },
                    "type": "a",
                  },
                  "Cras sit amet nibh libero, in gravida nulla (...)",
                ],
                "type": "div",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<UnFollowButton /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UnFollowButton
    followedUsername="cdvx"
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="btn  btn-outline-blue btn-lg"
        href="#"
      >
        Unfollow 
        cdvx
      </a>,
      "className": "blog-pagination",
      "id": "unfollow",
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Unfollow ",
          "cdvx",
        ],
        "className": "btn  btn-outline-blue btn-lg",
        "href": "#",
      },
      "ref": null,
      "rendered": Array [
        "Unfollow ",
        "cdvx",
      ],
      "type": "a",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="btn  btn-outline-blue btn-lg"
          href="#"
        >
          Unfollow 
          cdvx
        </a>,
        "className": "blog-pagination",
        "id": "unfollow",
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Unfollow ",
            "cdvx",
          ],
          "className": "btn  btn-outline-blue btn-lg",
          "href": "#",
        },
        "ref": null,
        "rendered": Array [
          "Unfollow ",
          "cdvx",
        ],
        "type": "a",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
